pkg/services/services_rawtx_test.go

```diff
@@ -165,4 +193,92 @@ func TestRawTxFailure(t *testing.T) {
 		assert.Error(t, err)
 		assert.Contains(t, err.Error(), "all services failed")
 	})
+	t.Run("Bitails returns error when raw tx hex is malformed", func(t *testing.T) {
+		// given:
+		given := testservices.GivenServices(t)
+		txID := "abc123"
+		given.Bitails().WillReturnRawTxHex(txID, "bad-$$$-hex")
+
+		services := given.Services().WithDefaultConfig()
+
+		// when:
+		_, err := services.RawTx(txID)
+
+		// then:
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "decode hex failed")
+		assert.Contains(t, err.Error(), "Bitails")
+	})
+
+}
+
+func TestWalletServices_RawTx_ErrorCases(t *testing.T) {
+	txID := "3c64c621c0070ea56ca2ef13ef699483c3938f48e030b184f1d094678eda7ab8"
+	malformedHex := "illegal-%-hex-char-$"
+
+	tests := []struct {
+		name                 string
+		setup                func(ts.ServicesFixture)
+		expectedErrorMessage string
+	}{
+		{
+			name: "WOC unreachable - Bitails returns malformed hex",
+			setup: func(f ts.ServicesFixture) {
+				_ = f.WhatsOnChain().WillBeUnreachable()
```

Comment:
```suggestion
				f.WhatsOnChain().WillBeUnreachable()
```
or
```suggestion
				err := f.WhatsOnChain().WillBeUnreachable()
				require.NoError(t, err)
```

and same in other tests

Reply:
@jakubmkowalski  why i need to check error here if i am mocking unreachable behaving .. and these mocks always return errors ... and in the last case we simulating both instances to return error as unreachable, and by invoking Unreachable u failing WoC response  .. so error is required ...

Reply:
however i added check to make sure we getting error there `require.Error(t, err)`